vowpalwabbit/gd.h

```diff
@@ -49,12 +49,22 @@ inline void vec_add_multipredict(multipredict_info<T>& mp, const float fx, uint6
       p->scalar +=
           fx * mp.weights[i];  // TODO: figure out how to use weight_parameters::iterator (not using change_begin())
   }
-  else  // TODO: this could be faster by unrolling into two loops
-    for (size_t c = 0; c < mp.count; ++c, fi += (uint64_t)mp.step, ++p)
+  else
+  {
+    uint64_t count = 0;
+    uint64_t copy_fi = (fi + (uint64_t)mp.step) & mask;
+    for (; copy_fi != fi && count < mp.count; copy_fi += (uint64_t)mp.step)
+    {
+      copy_fi &= mask;
+      ++count;
+    }
```

Comment:
Including line number 59 in line number 56 as below would be better since that is the convention being followed in neighbour for loops.

`for (; copy_fi != fi && count < mp.count; copy_fi += (uint64_t)mp.step, ++count)`